hollow/src/main/java/com/netflix/hollow/api/producer/validation/ValidationStatus.java

```diff
@@ -0,0 +1,151 @@
+ *
+ */
+public class ValidationStatus {
+    private final long version;
+    private final Status status;
+    private final Throwable throwable;
+    private final HollowProducer.ReadState readState;
+    private final List<ValidatorStatus> validatorStatusList;
+
+	@Generated("SparkTools")
+	private ValidationStatus(Builder builder) {
+		this.version = builder.version;
+		this.status = builder.status;
+		this.throwable = builder.throwable;
+		this.readState = builder.readState;
+		this.validatorStatusList = builder.validatorStatusList;
+	}
+    
+	public ValidationStatus(long version, Status status, Throwable throwable, ReadState readState,
+			List<ValidatorStatus> validatorStatusList) {
+		super();
+		this.version = version;
+		this.status = status;
+		this.throwable = throwable;
+		this.readState = readState;
+		this.validatorStatusList = validatorStatusList;
+	}
+
+	public long getVersion() {
+		return version;
+	}
+
+	public Status getStatus() {
+		return status;
+	}
+
+	public Throwable getThrowable() {
+		return throwable;
+	}
+
+	public HollowProducer.ReadState getReadState() {
+		return readState;
+	}
+
+	public List<ValidatorStatus> getValidatorStatusList() {
+		return validatorStatusList;
+	}
+
+	@Override
+	public String toString() {
+		return "ValidationStatus [version=" + version + ", status=" + status + ", throwable=" + throwable
+				+ ", validatorStatusList=" + validatorStatusList + "]";
+	}
+
+	/**
+	 * Creates builder to build {@link ValidationStatus}.
+	 * @return created builder
+	 */
+	@Generated("SparkTools")
+	public static Builder builder() {
+		return new Builder();
+	}
+
+	/**
+	 * Builder to build {@link ValidationStatus}.
+	 */
+	@Generated("SparkTools")
+	public static final class Builder {
+		private long version;
+		private Status status;
+		private Throwable throwable;
+		private HollowProducer.ReadState readState;
+		private List<ValidatorStatus> validatorStatusList;
+
+		private Builder() {
+			validatorStatusList = new ArrayList<>();
+		}
+
+		public Builder withVersion(long version) {
+			this.version = version;
+			return this;
+		}
+
+		public Builder success() {
+			this.status = Status.SUCCESS;
+			return this;
+		}
+		
+		public Builder fail(Throwable th) {
+			this.status = Status.FAIL;
+			this.throwable = th;
+			return this;
+		}
+
+		public Builder withThrowable(Throwable throwable) {
+			this.throwable = throwable;
+			return this;
+		}
+
+		public Builder withReadState(HollowProducer.ReadState readState) {
+			this.readState = readState;
+			return this;
+		}
+
+		public Builder addValidatorStatus(Throwable th, String message) {
+			Status success = (th == null)? Status.SUCCESS: Status.FAIL;
+			this.validatorStatusList.add(ValidatorStatus.builder().withStatus(success).withMessage(message).withThrowable(th).withVersion(version).build());
+			return this;
+		}
+
+		public ValidationStatus build() {
+			return new ValidationStatus(this);
```

Comment:
Not a big fan of circular dependency ValidationStatus -> Builder -> ValidationStatus

Reply:
It is pure builder pattern as far as I remember. Builder is static. Let me recheck.

Reply:
Confirmed that there is no circular dependency. It is a static builder returning an object.